### PR TITLE
testing: non-existent symlink on demand

### PR DESCRIPTION
**Pull request by flowlo**
Fixes fsouza/go-dockerclient#368

### CODE REVIEW

File: build_test.go [hunk 1]
```diff
@@ -35,6 +35,11 @@ func TestBuildImageMultipleContextsError(t *testing.T) {
 func TestBuildImageContextDirDockerignoreParsing(t *testing.T) {
 	fakeRT := &FakeRoundTripper{message: "", status: http.StatusOK}
 	client := newTestClient(fakeRT)
+
+	if err := os.Symlink("doesnotexist", "testing/data/symlink"); err != nil {
+		t.Errorf("error creating symlink on demand: %s", err)
+	}
+
```
Comment (fsouza) [inline build_test.go]: There are some calls to t.Fatal in this test, so I'd prefer to have a defer just after this if:

```
defer func() {
	if err := os.Remove("testing/data/symlink"); err != nil {
		t.Errorf("error removing symlink on demand: %s", err)
	}
}()
```

Thank you!

Comment (flowlo) [inline build_test.go]: Oh sure!

**Comment by flowlo** [comment]
There is one failing job, but I can't figure out how this correlates with my changes.

**Comment by flowlo** [comment]
Ping @fsouza @peter-edge

**Comment by peter-edge** [comment]
The defer will not go off if any t.Fatal function is hit. Therefore, this might not get cleaned up. That's OK, but I think you should handle it, both in the .gitignore, and check for existence before creating it.

**Comment by flowlo** [comment]
Adding `testing/data/symlink` to `.gitignore` makes sense, but I am not so sure about adding logic to handle any existing file. Probably we'd have to `os.Lstat` the symlink. In this case (a) `os.Lstat` itself can fail, (b) the file could exists but not be a symlink, and (c) should we remove it?

Do you really think that this will be the case very often? I'd probably let the test error if there's a file named `testin/data/symlink` and leave the rest up to the user executing the tests.

**Comment by fsouza** [comment]
@peter-edge yeah, you're right about the fatal vs defer thing :-(

@flowlo I think it's fine to merge as is, maybe we can drop Fatal usage in the future. Thanks for contributing!

**Pull request merged**